scipy/interpolate/_bsplines.py

```diff
@@ -445,34 +476,87 @@ def integrate(self, a, b, extrapolate=None):
         if extrapolate is None:
             extrapolate = self.extrapolate
 
-        if not extrapolate:
-            # shrink the integration interval, if needed
+        # Prepare self.t and self.c.
+        self._ensure_c_contiguous()
+
+        # Swap integration bounds if needed.
+        sign = 1
+        if b < a:
+            a, b = b, a
+            sign = -1
+        n = self.t.size - self.k - 1
+
+        if not extrapolate and (extrapolate !="periodic"):
```

Comment:
I believe this better should be written as 

```if extrapolate != "periodic" and not extrapolate```, because it is not apparent to what `not "periodic"` (for example) evaluates.